freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForStackService.java

```diff
@@ -96,7 +100,11 @@ public SyncStatusDetail synchronizeStack(Stack stack, UmsUsersState umsUsersStat
                 }
             }
 
-            return toSyncStatusDetail(environmentCrn, warnings);
+            SyncStatusDetail syncStatusDetail = toSyncStatusDetail(environmentCrn, warnings);
+            if (SynchronizationStatus.COMPLETED.equals(syncStatusDetail.getStatus())) {
+                authDistributorService.updateAuthViewForEnvironment(environmentCrn, umsUsersState, stack.getAccountId());
```

Comment:
We need to make sure or prevent the filtering of user and machineuser's since I see that here we are filling cnrs before creating UmsStatus. 
http://github.com/hortonworks/cloudbreak/blob/38af0db40eecf795ffab678db59fffe1d4bfa480/freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/ums/DefaultUmsUsersStateProvider.java#L73-L73